### ISSUE TITLE
SMTP TLS on Windows not working (hang forever)

### ISSUE THREAD

**Issue by Maloupi** [PHPMailer/PHPMailer#2958]
# Problem description
On windows (PHP 7.4 -> PHP 8.2) (no problem on Linux), sending a mail via SMTP with TLS (no problem with SSL) hang forever on **stream_socket_enable_crypto** in **startTLS()** in **SMTP.php**

I think this is because it use tls 1.0 first.
if i force crypto_method to tls 1.1 or 1.2, it works well :
```
$crypto_method = STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
```

# Code to reproduce
```<?php

define('SMTP_SERVER', '**********');
define('SMTP_PORT', '587');
define('SMTP_ENCRYPTION', 'tls');
define('SMTP_ACCOUNT', '**********');
define('SMTP_PASSWORD', '**********');
define('DEST_ADDRESS', '**********');

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require 'phpmailersrc/Exception.php';
require 'phpmailersrc/PHPMailer.php';
require 'phpmailersrc/SMTP.php';

$mail = new PHPMailer(true);

try {
    $mail->isSMTP();                      
    $mail->Host       = SMTP_SERVER;      
    $mail->SMTPAuth   = true;             
    $mail->Username   = SMTP_ACCOUNT;     
    $mail->Password   = SMTP_PASSWORD;    
    $mail->SMTPSecure = SMTP_ENCRYPTION;  
    $mail->Port       = SMTP_PORT;    
    
    $mail->setFrom(SMTP_ACCOUNT);
    $mail->addAddress(DEST_ADDRESS);
    $mail->isHTML(true);    
    $mail->Subject = 'Here is the subject';
    $mail->Body    = 'This is the HTML message body <b>in bold!</b>';
    $mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

    $mail->send();
    echo 'Message has been sent';
} catch (Exception $e) {
    echo "Message could not be sent. Mailer Error: {$mail->ErrorInfo}";
}
```

**Comment by Synchro** [comment]
TLS protocols don't work like that; it doesn't try different versions in order, but presents all the options as a list. It may be that your Windows machine has a buggy TLS implementation that breaks on some configs. There isn't a built-in way to change this, but you can achieve it by subclassing the SMTP class (like in [this example](https://github.com/PHPMailer/PHPMailer/blob/master/examples/smtp_low_memory.phps)), overriding the `startTLS` method, and then injecting your instance back into PHPMailer using `setSMTPInstance()`.

Short version:

```php
class mySMTP extends SMTP
{
    public function startTLS()...
}

$mail = new PHPMailer();
$mail->setSMTPInstance(new mySMTP());
```

You can then carry on as normal.

*Note to myself; I need to update the override example to use `setSMTPInstance()`*

**Comment by Maloupi** [comment]
Thank you
